Anagram.java

```diff
@@ -2,48 +2,98 @@
 public class Anagram {
 	public static void main(String args[]) {
 		// Tests the isAnagram function.
-		System.out.println(isAnagram("silent","listen"));  // true
-		System.out.println(isAnagram("William Shakespeare","I am a weakish speller")); // true
-		System.out.println(isAnagram("Madam Curie","Radium came")); // true
-		System.out.println(isAnagram("Tom Marvolo Riddle","I am Lord Voldemort")); // true
+		System.out.println(isAnagram("silent", "listen")); // true
+		System.out.println(isAnagram("William Shakespeare", "I am a weakish speller")); // true
+		System.out.println(isAnagram("Madam Curie", "Radium came")); // true
+		System.out.println(isAnagram("Tom Marvolo Riddle", "I am Lord Voldemort")); // true
 
 		// Tests the preProcess function.
 		System.out.println(preProcess("What? No way!!!"));
-		
+
 		// Tests the randomAnagram function.
 		System.out.println("silent and " + randomAnagram("silent") + " are anagrams.");
-		
-		// Performs a stress test of randomAnagram 
+
+		// Performs a stress test of randomAnagram
 		String str = "1234567";
-		Boolean pass = true;
+		boolean pass = true;
 		//// 10 can be changed to much larger values, like 1000
 		for (int i = 0; i < 10; i++) {
 			String randomAnagram = randomAnagram(str);
 			System.out.println(randomAnagram);
 			pass = pass && isAnagram(str, randomAnagram);
-			if (!pass) break;
+			if (!pass)
+				break;
 		}
 		System.out.println(pass ? "test passed" : "test Failed");
-	}  
+	}
 
 	// Returns true if the two given strings are anagrams, false otherwise.
 	public static boolean isAnagram(String str1, String str2) {
-		// Replace the following statement with your code
-		return false;
+		str1 = removeSpace(preProcess(str1));
+		str2 = removeSpace(preProcess(str2));
+		if (str1.length() != str2.length())
+			return false;
+
+		for (int i = 0; i < str1.length(); i++) {
+			int count1 = 0;
+			int count2 = 0;
+			char c = str1.charAt(i);
+			for (int j = 0; j < str1.length(); j++) {
+				if (str1.charAt(j) == c)
+					count1++;
+				if (str2.charAt(j) == c)
+					count2++;
+			}
+			if (count1 != count2)
+				return false;
+
+		}
+		return true;
 	}
```

Comment:
PERFECT. 

But I do have a suggestion.. 
you are double counting, however once finished counting letters once, the is no need for another check. 
consider 2 strings with length 1000 such that both are equal "aaaaaaaaa....ada...a"

consider the following code: 
```
for (int i = 0; i < str1.length(); i++) {
	int count1 = 0;
	int count2 = 0;
	char c = str1.charAt(i);
	// checks if i is the first index char c appears is, if not (and still in loop it is valid therefore skipped)
	// otherwise regular check.
	boolean isFirstAppearanceOfChar = str1.indexOf(c) == i;
	if (!isFirstAppearanceOfChar) {
		continue;
	}
	for (int j = 0; j < str1.length(); j++) {
		if (str1.charAt(j) == c){
			count1++;
		}
		if (str2.charAt(j) == c) {
			count2++;
		}
	}
	if (count1 != count2) {
		return false;
	}
}
```